tests/kernel/semaphore/sys_sem/src/main.c

```diff
@@ -33,16 +33,6 @@ static ZTEST_DMEM atomic_t atomic_count;
  * @}
  */
 
-static void sem_thread_give_uninit(void *p1, void *p2, void *p3)
-{
-	ztest_set_fault_valid(true);
-
-	/* use sem without initialise */
-	k_sem_give(&uninit_sem);
```

Comment:
In the docs we have:

"A semaphore must be initialized before it can be used."

This was also is part of the requirements currently being worked on by the safety WG and that test was targeting this requirement and the way it is documented, so the test actually looks valid to me.

Reply:
If that is the requirement, then our existing k_sem implementation is already failing. :)

I think you're misunderstanding the test code: it's not checking k_sem_give() against an invalid input and verifying it correctly handles the error.  It's calling k_sem_give() on an uninitialized semaphore **and expecting it to crash** (i.e. you need to abort the thread via ztest_set_fault_valid() or else you reach the ztest_test_fail() on the next line).  In fact, k_sem_give() does **not** handle the case of invalid input, it blindly assumes it's initialized and then crashes trying to mutate the null pointers in the waitq dlist.

The test is literally verifying the opposite of that requirement.  It needs to go.  I'm very happy to add validity checking to zync though if you want that.